src/scripts/actions/UserActions.js

```diff
@@ -4,27 +4,97 @@ import { createAction } from 'redux-actions'
 import Promise from 'bluebird'
 import Cookies from 'js-cookie'
 
-import { LOGIN_REQUESTED, LOGIN_SUCCESS, LOGOUT } from 'constants/ActionConstants'
+import {
+  LOGIN_REQUESTED,
+  LOGIN_SUCCESS,
+  LOGOUT,
+  SET_WALLET_DATA
+} from 'constants/ActionConstants'
+import paratii from 'utils/ParatiiLib'
+import { DEFAULT_PASSWORD } from 'constants/ParatiiLibConstants'
+import { getWalletKey, getMnemonicKey } from 'selectors/index'
 
-import type { Dispatch } from 'redux'
+import type { RootState, Dispatch } from 'types/ApplicationTypes'
 
 const loginRequested = createAction(LOGIN_REQUESTED)
 const loginSuccess = createAction(LOGIN_SUCCESS)
 const logoutAction = createAction(LOGOUT)
+const setWalletData = createAction(SET_WALLET_DATA)
 
-const sleep = (ms) => {
+const sleep = ms => {
   return new Promise(resolve => setTimeout(resolve, ms))
 }
 
-export const login = (email: string, password: string) => (dispatch: Dispatch<*>) => {
+export const login = (email: string, password: string) => (
+  dispatch: Dispatch
+) => {
   dispatch(loginRequested())
   sleep(200).then(() => {
     Cookies.set('email', email)
-    dispatch(loginSuccess({email}))
+    dispatch(loginSuccess({ email }))
   })
 }
 
-export const logout = () => (dispatch: Dispatch<*>) => {
-  dispatch(logoutAction())
+export const logout = () => (dispatch: Dispatch) => {
   Cookies.remove('email')
+  dispatch(logoutAction())
+}
+
+const setAndSyncWalletData = ({
+  walletKey,
+  wallet,
+  mnemonic,
+  mnemonicKey
+}: {
+  walletKey: string,
+  wallet: Object,
+  mnemonicKey: string,
+  mnemonic: string
+}) => (dispatch: Dispatch): void => {
+  dispatch(
+    setWalletData({
+      walletKey,
+      mnemonicKey
+    })
+  )
+  localStorage.setItem(walletKey, JSON.stringify(wallet))
+
+  // TODO: needs to be encrypted
+  localStorage.setItem(mnemonicKey, mnemonic)
+}
+
+export const setupKeystore = () => async (
+  dispatch: Dispatch,
+  getState: () => RootState
+) => {
+  paratii.eth.wallet.clear()
+  const walletKey: string = getWalletKey(getState())
+  const mnemonicKey: string = getMnemonicKey(getState())
+  const walletString: string = localStorage.getItem(walletKey) || ''
+  let wallet: ?Object
+  let mnemonic: string = localStorage.getItem(mnemonicKey) || ''
+  let walletIsValid: boolean = true
+  if (walletString) {
+    try {
+      wallet = paratii.eth.wallet.decrypt(
+        JSON.parse(walletString),
+        DEFAULT_PASSWORD
+      )
+    } catch (err) {
+      walletIsValid = false
+    }
+  }
+  if (!wallet || !walletIsValid) {
+    paratii.eth.wallet.create()
+    mnemonic = await paratii.eth.wallet.getMnemonic()
+    wallet = paratii.eth.wallet.encrypt(DEFAULT_PASSWORD)
+    dispatch(
+      setAndSyncWalletData({
+        wallet,
+        walletKey,
+        mnemonic,
+        mnemonicKey
+      })
+    )
+  }
```

Comment:
@jellegerbrandy I refactored this logic a bit (and moved it to this `UserActions.js` module). 

The idea behind the change was:
- move it here because this is asynchronous logic that affects our state; thus, it should be an action
- this needed to be more connected to our state; now, the action tries to grab the correct keys from the store and if it does not get them, it creates a new wallet and updates the store